pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -343,98 +579,89 @@ func CompactNewChunks(ctx context.Context, series Series, bt *v1.BloomTokenizer,
+		// There must be a better way to get this, ordering chunkRefs by timestamp doesn't fully solve it
+		// chunk files name have this info in ObjectStore, but it's not really exposed
+		minFrom := model.Latest
+		maxThrough := model.Earliest
 
-	for _, s := range series {
-		err := storeClient.indexShipper.ForEach(ctx, s.tableName, s.tenant, func(isMultiTenantIndex bool, idx shipperindex.Index) error {
-			if isMultiTenantIndex {
-				return nil
+		for _, c := range chks {
+			if minFrom > c.From {
+				minFrom = c.From
+			}
+			if maxThrough < c.From {
+				maxThrough = c.Through
 			}
+		}
 
-			// TODO make this casting safe
-			_ = idx.(*tsdb.TSDBFile).Index.(*tsdb.TSDBIndex).ForSeries(
-				ctx,
-				nil,              // Process all shards
-				0, math.MaxInt64, // Replace with MaxLookBackPeriod
-
-				// Get chunks for a series label and a fp
-				func(ls labels.Labels, fp model.Fingerprint, chksMetas []tsdbindex.ChunkMeta) {
-
-					// TODO call bloomShipperClient.GetMetas to get existing meta.json
-					var metas []bloomshipper.Meta
-
-					if len(metas) == 0 {
-						// Get chunks data from list of chunkRefs
-						chks, err := storeClient.chunk.GetChunks(
-							ctx,
-							makeChunkRefs(chksMetas, s.tenant, fp),
-						)
-						if err != nil {
-							level.Info(util_log.Logger).Log("getting chunks", err)
-							return
-						}
-
-						// effectively get min and max of timestamps of the list of chunks in a series
-						// There must be a better way to get this, ordering chunkRefs by timestamp doesn't fully solve it
-						// chunk files name have this info in ObjectStore, but it's not really exposed
-						minFrom := model.Latest
-						maxThrough := model.Earliest
-
-						for _, c := range chks {
-							if minFrom > c.From {
-								minFrom = c.From
-							}
-							if maxThrough < c.From {
-								maxThrough = c.Through
-							}
-						}
-
-						series := Series{
-							tableName:   s.tableName,
-							tenant:      s.tenant,
-							labels:      ls,
-							fingerPrint: fp,
-							chunks:      chks,
-							from:        minFrom,
-							through:     maxThrough,
-							indexPath:   s.indexPath,
-						}
-
-						err = CompactNewChunks(ctx, series, bt, bloomShipperClient, c.cfg.WorkingDirectory)
-						if err != nil {
-							return
-						}
-					} else {
-						// TODO complete part 2 - periodic compaction for delta from previous period
-						// When already compacted metas exists
-						// Deduplicate index paths
-						uniqueIndexPaths := make(map[string]struct{})
-
-						for _, meta := range metas {
-							for _, blockRef := range meta.Blocks {
-								uniqueIndexPaths[blockRef.IndexPath] = struct{}{}
-								//...
-							}
-						}
-
-					}
-				})
-			return nil
-		})
+		series := Series{
+			tableName:   job.tableName,
+			tenant:      job.Tenant(),
+			labels:      job.Labels(),
+			fingerPrint: job.Fingerprint(),
+			chunks:      chks,
+			from:        minFrom,
+			through:     maxThrough,
```

Comment:
suggestion: we can get `from` and `through` from chunkMetas. 
and you don't necessarily have to have chunks here in this struct. This way you can remove this and use Job instead. Pass the chunks to `CompactNewChunks` separately. 
`CompactNewChunks` will not always compact all chunks returned. Once completed, we will have another logic before calling `CompactNewChunks` to filter only the chunks that are not been compacted into blooms before.

Reply:
Done, thank you!